checkmates/data_checks/checks/distribution_data_check.py

```diff
@@ -0,0 +1,157 @@
+            ...         "action_options": []
+            ...     }
+            ... ]
+        """
+        messages = []
+
+        if y is None:
+            messages.append(
+                DataCheckError(
+                    message="Target is None",
+                    data_check_name=self.name,
+                    message_code=DataCheckMessageCode.TARGET_IS_NONE,
+                    details={},
+                ).to_dict(),
+            )
+            return messages
+
+        y = infer_feature_types(y)
+        allowed_types = [
+            ww.logical_types.Integer.type_string,
+            ww.logical_types.Double.type_string,
+            ww.logical_types.Age.type_string,
+            ww.logical_types.AgeFractional.type_string,
+        ]
+        is_supported_type = y.ww.logical_type.type_string in allowed_types
+
+        if not is_supported_type:
+            messages.append(
+                DataCheckError(
+                    message="Target is unsupported {} type. Valid Woodwork logical types include: {}".format(
+                        y.ww.logical_type.type_string,
+                        ", ".join([ltype for ltype in allowed_types]),
+                    ),
+                    data_check_name=self.name,
+                    message_code=DataCheckMessageCode.TARGET_UNSUPPORTED_TYPE,
+                    details={"unsupported_type": y.ww.logical_type.type_string},
+                ).to_dict(),
+            )
+            return messages
+
+        (
+            is_skew,
+            distribution_type,
+            skew_value,
+            coef,
+        ) = _detect_skew_distribution_helper(y)
+
+        if is_skew:
+            details = {
+                "distribution type": distribution_type,
+                "Skew Value": skew_value,
+                "Bimodal Coefficient": coef,
+            }
+            messages.append(
+                DataCheckWarning(
+                    message="Target may have a skewed distribution.",
+                    data_check_name=self.name,
+                    message_code=DataCheckMessageCode.TARGET_SKEWED_DISTRIBUTION,
+                    details=details,
+                    action_options=[
+                        DataCheckActionOption(
+                            DataCheckActionCode.TRANSFORM_TARGET,
+                            data_check_name=self.name,
+                            metadata={
+                                "is_target": True,
+                                "transformation_strategy": "yeojohnson",
+                            },
+                        ),
+                    ],
+                ).to_dict(),
+            )
+        return messages
+
+
+def _detect_skew_distribution_helper(y):
+    """Helper method to detect skewed or bimodal distribution. Returns boolean, distribution type, the skew value, and bimodal coefficient."""
+    skew_value = np.stats.skew(y)
+    coef = diptest.diptest(y)[1]
+
+    if coef < 0.05:
```

Comment:
So the `diptest` takes precedence over skewness?

Reply:
yeah, this is because the diptest is testing for bimodal distributions, which can show up with a skew of zero, which wouldn't trigger the error even though it is a distribution we are looking to raise a warning for